src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -250,6 +256,11 @@ public abstract partial class JsonNamingPolicy
     public partial class JsonNode
     {
         public JsonNode() { }
+        public System.Text.Json.JsonElement AsJsonElement() { throw null; }
```

Comment:
Not related to this change in particular, but I supose we want JsonNode class to be abstract?

Reply:
... and possibly with a `private protected` constructor :smile:.

Reply:
private protected? Not sure I got why...

Reply:
So that the type can't be extended outside this assembly.  (It doesn't really make sense for a 3rd party library to define a new kind of JsonNode, and even if they did it probably wouldn't work very well with JsonElement)

Reply:
I see, I thought that would be protected internal (child classes can call it as long as they are internal) but I know that there is some complicated explanation of why protected internal and private protected are very similar except one is a logical AND and the other one is an OR...

Reply:
`protected internal` is "protected OR internal" (IL: famOrAssm), `private protected` is new, and what many people thought `protected internal` meant ("protected AND internal") (IL: famAndAssm).

The only real mnemonic is to remember that both `private` and `internal` can be applied as modifiers to `protected`, and that `private` means a smaller scope than `internal`.